app/views/operatingsystems/_form.html.erb

```diff
@@ -22,6 +22,11 @@
       <%= text_f f, :description, :help_inline => _("OS friendly name; e.g. RHEL 6.5") %>
       <%= select_f f, :family, Operatingsystem.families_as_collection, :value, :name, { :include_blank => _("Choose a family") }, { :label => _("Family"), :onchange => 'show_release(this);' } %>
       <div id="release_name" <%= display?(!@operatingsystem.use_release_name?) %>>
```

Comment:
`@operatingsystem` will be an `OperatingSystem` object before save, so you can change the use_release_name? function to something like:
```ruby
  def use_release_name?
    return false unless family
    self.becomes(family.constantize).use_release_name?
  end
```

Reply:
Thanks Tomer. Very appreciated. I added one more line to prevent a loop and therefore not to run in a stack level too deep.